### ISSUE TITLE
qss integration failed to configure after homeassitant upgrade to 2024.12.0

### ISSUE THREAD

**Issue by gco33** [CM000n/qss#52]
**Describe the bug**
qss integration failed to configure after homeassitant upgrade to 2024.12.0
exact message in log is "Setup failed for custom integration 'qss': Requirements for qss not found: ['questdb>=1.2,<2.0']."

**To Reproduce**
Once homeassistant have been upgraded and restarted, notification is raised with message above

**Expected behavior**
qss should start without error after upgrade

Exact version of each module of homeassistant :
![image](https://github.com/user-attachments/assets/c89c3ad2-0c39-4890-8f6d-d50fb824fdb9)
Interface utilisateur (in french) is User interface

**Comment by FeHJa** [comment]
I see the following line in the logs:

> 2024-12-05 13:22:31.569 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration 'qss': Requirements for qss not found: ['questdb>=1.2,<2.0'].

**Comment by antst** [comment]
Yes, this is bug which everyone with supervised  2024.12.0 will experience.
Tried to dig this issue couple of days ago, and it all boils down to very strange error of being unable to load cpython shared library.
I've found only two similar issue in the google history, both miss resolutions.

**Comment by antst** [comment]
But, technically, this is not QSS issue, but rather issue of questdb python package.

**Comment by antst** [comment]
I created issue on HA core repo, where we need to work on resolution.
https://github.com/home-assistant/core/issues/132398

**Comment by gco33** [comment]
thanks a lot for investigation

**Comment by antst** [comment]
OK, why it does not build on new HA core I sorted out.
But, aside issue with /tmp on HA core, there are ton of issues with questdb module.
1) versions bellow 2.0.3  has bugs in build system. so pip will not build and install them "as is", unless some fixed applied.
2) even if fixes applied, it need Rust compiler to build. While there is no problem with build of 2.0.x versions . 1.2.0 has some other problems with rust. And current extension works with 1.2.0 only.

So, outcome:
Either some will upgrade this component to be able to work with 2.0.3 (but then still building of questb on the fly by HA Core would require installed Rust compiler and fixed permission of /tmp) or someone takes car of generation of pre-built wheel for environment we have in new HA Core.  Or ideally, someone update compinent to use 2.0.3 questdb package and generate wheel :)

To be honest, I have ZERO experience with wheels etc.

**Comment by antst** [comment]
OK,  #53 updates qss to use lates questdb python client.  Client which can be built by pip (assuming /tmp is not mounted with "noexec" and rust compiler "cargo" is installed on the host.
In case of hass core it is simple matter of 
```
mount -t tmpfs -o rw,nosuid,nodev,relatime tmpfs /tmp
apk add cargo
pip install -v questdb==2.0.3
```

I will also talk to questdb people to provide prebuilt wheel  for this particular combination of python and musl. But, before wheel  is available, functionality already can be restored with couple of extra command above

**Comment by FeHJa** [comment]
Thank you @antst, works for me.

**Comment by antst** [comment]
Update: 
HA team added wheel for questdb=2.0.3  module.
So, after next upgrade it should roll fine with current qss integration, which is updated to use 2.0.3.
